### PR TITLE
Reorganize `omf` plugin structure

### PR DESCRIPTION
**Pull request by derekstavis**
This updates omf plugin to new architecture, moving
functions to `functions` directory. It also revamps the
separation of concerns between CLI and underlying
implementations, organizing them into directories,
whose are autoloaded in plugin startup.

### CODE REVIEW

File: pkg/omf/functions/omf.fish [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    set command $argv[1]
+    set arguments
+  else
+    set command help
+    set arguments
+  end
+
+  switch "$command"
+    case "-v*" "--v*"
+      omf.version
```
Comment (bpinto) [inline pkg/omf/functions/omf.fish]: Omf.cli.version

File: pkg/omf/functions/omf.fish [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    case "doctor"
+      omf.doctor
```
Comment (bpinto) [inline pkg/omf/functions/omf.fish]: Omf.cli.doctor

**Comment by rrosson** [comment]
The only issue I see with this is what if user defined completions and functions are already in ~/.config/omf? I think in the past this was the recommended place to put these directories and have them called out from the init.fish in the same directory.

If the plan is to move forward this will need to be considered.

File: pkg/omf/functions/packages/omf.packages.remove.fish [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      or continue
+
+    test $pkg = (cat $OMF_CONFIG/theme);
+      and echo default > $OMF_CONFIG/theme
+
+    if rm -rf $path
+      omf.bundle.remove "theme" $pkg
+      return 0
+    else
+      return 1
```
Comment (bpinto) [inline pkg/omf/functions/packages/omf.packages.remove.fish]: We don't use these return value.

File: pkg/omf/functions/packages/omf.packages.update.fish [hunk 4]
```diff
@@ -1,11 +1,14 @@
-function omf.update -a name
-  function __omf.update.success
-    echo (omf::em)"✔ $argv successfully updated."(omf::off)
-  end
+function __omf.packages.update.success
+  echo (omf::em)"✔ $argv successfully updated."(omf::off)
+  return 0
```
Comment (bpinto) [inline pkg/omf/functions/packages/omf.packages.update.fish]: Everywhere else we have these helper functions we dont put the return value inside of the function.

File: pkg/omf/functions/packages/omf.packages.remove.fish (hunk 3 above)
Comment (derekstavis) [inline pkg/omf/functions/packages/omf.packages.remove.fish]: They are now used on cli to distinguish package not found from rm failing

File: pkg/omf/functions/packages/omf.packages.update.fish (hunk 4 above)
Comment (derekstavis) [inline pkg/omf/functions/packages/omf.packages.update.fish]: As they are the last function called by parent function their return values are propagated

File: pkg/omf/omf.bundle.install.fish [hunk 5]
```diff
@@ -0,0 +1 @@
+functions/bundle/omf.bundle.install.fish
```
Comment (bpinto) [inline pkg/omf/omf.bundle.install.fish]: ?

Comment (derekstavis) [inline pkg/omf/omf.bundle.install.fish]: It's a symlink :)

**Comment by derekstavis** [comment]
Upstreaming! :shipit:

**Pull request merged**